### DISCUSSION TITLE
Langchain version 0.1.xx

### DISCUSSION THREAD

**Discussion by pxz2016** [danny-avila/LibreChat#1685] [Q&A]
### What features would you like to see added?

Is there a plan to update Langchain to version 0.1.xx

### More details

Langchainjs 0.1.xx has been updated with many new features, such as:
convertToOpenAITool
executor.stream

Can enhance the experience of function calls

### Which components are impacted by your request?

Plugins

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Sure I think overall plug-ins will be removed and tools will be used by endpoints that support them.

Langchain still has some issues with streaming and the new LCEL: https://github.com/langchain-ai/langchainjs/issues/3398

Assistants API will be out first which will introduce better tool handling. Then I can focus on that

**Reply by pxz2016** [reply]
Assistants API!  When is it expected to be released?

**Reply by danny-avila** [reply]
Plan to do a soft-launch today and finalize things tomorrow. Join the discord to stay updated: https://discord.librechat.ai
